### PR TITLE
Add INSTALL.md with pointers to plugin and operator installation + new HELM instructions

### PR DESCRIPTION
**Pull request by tkatila**
In relation to issues #1312 and #1061

Signed-off-by: Tuomas Katila <tuomas.katila@intel.com>

### CODE REVIEW

**Comment by tkatila** [comment]
I'll rewrite this as a more generic installation instructions.

**Comment by eero-t** [comment]
> I'll rewrite this as a more generic installation instructions.

Any news?

**Comment by tkatila** [comment]
> Any news?

It's coming still. I just had other items to work with.

**Comment by tkatila** [comment]
In reference to #1061, I wasn't sure if these files should be also stripped from namespaces:
deployments/fpga_plugin/overlays/af/kustomization.yaml
deployments/sgx_admissionwebhook/overlays/default-with-certmanager/kustomization.yaml
deployments/sgx_plugin/overlays/epc-register/kustomization.yaml
deployments/sgx_plugin/overlays/epc-register/service-account.yaml

At least the _intelfpgaplugin-system_ namespace is referred in shell scripts and READMEs.

File: INSTALL.md [hunk 1]
```diff
@@ -0,0 +1,89 @@
+# Installing device plugins to cluster
+
+## Install device plugins via a daemonset
+
+Each plugin can be installed via a daemonset. The install changes slightly based on the desired plugin. See install instructions per [plugin](README.md#plugins).
+
+Installing plugins via daemonsets deployes them to the ```default``` (or currently active) namespace. Use kubectl's ```--namespace``` argument to change the deployment namespace.
```
Comment (eero-t) [inline INSTALL.md:7]: daemonset -> DaemonSet

See: https://kubernetes.io/docs/concepts/workloads/controllers/daemonset/

File: INSTALL.md [hunk 2]
```diff
@@ -0,0 +1,89 @@
...
+
+## Install device plugins via device plugin operator
+
+A more advanced install method is via device plugin operator. There the plugins are installed by supplying device plugin CRDs which the operator then converts into plugin deployments. See installation instructions in the [operator README](cmd/operator/README.md#installation).
```
Comment (eero-t) [inline INSTALL.md:11]: I think this:
> There the plugins are installed by supplying device plugin CRDs which the operator then converts into plugin deployments.

Would read better as:
> Operator configures plugin deployments based on the supplied device plugin CRDs (Custom Resource Definitions).

File: INSTALL.md [hunk 3]
```diff
@@ -0,0 +1,89 @@
...
+
+Device plugins installed via device plugin operator are installed in the device plugin namespace ```inteldeviceplugins-system```.
```
Comment (eero-t) [inline INSTALL.md:13]: More simply:
> Operator installs device plugins to the ```inteldeviceplugins-system``` namespace.

Btw. Is this overridable by CRD?   If yes, add "... by default".

File: INSTALL.md [hunk 4]
```diff
@@ -0,0 +1,89 @@
...
+
+## Install with HELM charts
+
+Device plugins can also be installed to a cluster with our Helm charts. Helm chart leverages the device-plugin operator which depends on cert-manager and NFD. The chart project for the operator is at https://github.com/intel/helm-charts/tree/main/charts/device-plugin-operator. Individual plugin projects are under https://github.com/intel/helm-charts/tree/main/charts/.
```
Comment (eero-t) [inline INSTALL.md:17]: Maybe:
> Device plugins can also be installed to a cluster using the [device operator Helm chart](https://github.com/intel/helm-charts/tree/main/charts/device-plugin-operator) (depending on cert-manager and NFD). Individual plugin projects can be installed with Helm charts in https://github.com/intel/helm-charts/tree/main/charts/.

File: INSTALL.md [hunk 5]
```diff
@@ -0,0 +1,89 @@
...
+
+NOTE: cert-manager install takes a while to complete.
+
+### Installing NFD
+
+```bash
+cat > tmp.yaml <<EOF
+master:
+  extraLabelNs: ["gpu.intel.com","sgx.intel.com"]
+  resourceLabels: ["gpu.intel.com/millicores","gpu.intel.com/memory.max","gpu.intel.com/tiles","sgx.intel.com/epc"]
```
Comment (eero-t) [inline INSTALL.md:49]: Don't other plugins provide resources that should be listed here?

Comment (tkatila) [reply INSTALL.md:49]: I couldn't find others. @mythi can you confirm?

File: INSTALL.md (hunk 3 above)
Comment (tkatila) [reply INSTALL.md:13]: CRD doesn't have a field for namespace, so no?

Comment (tkatila) [reply INSTALL.md:13]: Yep, it's up to the chart to handle namespaces. For example, prometheus can be installed as a dependency into a different namespace. But you'll have to supply the ns via values, not via helm install arguments.

File: INSTALL.md (hunk 5 above)
Comment (mythi) [reply INSTALL.md:49]: just GPU and SGX

File: INSTALL.md [hunk 6]
```diff
@@ -0,0 +1,84 @@
...
+
+Each plugin can be installed via a DaemonSet. The install changes slightly based on the desired plugin. See install instructions per [plugin](README.md#plugins).
+
+Installing plugins via DaemonSets deployes them to the ```default``` (or currently active) namespace. Use kubectl's ```--namespace``` argument to change the deployment namespace.
+
+## Install device plugins via device plugin operator
+
+A more advanced install method is via device plugin operator. Operator configures plugin deployments based on the supplied device plugin CRDs (Custom Resource Definitions). See installation instructions in the [operator README](cmd/operator/README.md#installation).
+
+Operator installs device plugins to the ```inteldeviceplugins-system``` namespace.
```
Comment (mythi) [inline INSTALL.md:13]:
```suggestion
Operator installs device plugins to the same namespace where the operator itself is deployed.
```

File: INSTALL.md [hunk 7]
```diff
@@ -0,0 +1,84 @@
...
+### Installing NFD
+
+```bash
+helm install nfd nfd/node-feature-discovery \
+  --namespace node-feature-discovery --create-namespace --version 0.12.1 \
+  --set 'master.extraLabelNs={gpu.intel.com,sgx.intel.com}' \
+  --set 'master.resourceLabels={gpu.intel.com/millicores,gpu.intel.com/memory.max,gpu.intel.com/tiles,sgx.intel.com/epc}'
+```
+
+### Installing operator
```
Comment (mythi) [inline INSTALL.md:52]: We will need a subsection that explains how to list available versions (with `--devel` helm param).

File: INSTALL.md [hunk 8]
```diff
@@ -0,0 +1,84 @@
...
+
+A more advanced install method is via device plugin operator. Operator configures plugin deployments based on the supplied device plugin CRDs (Custom Resource Definitions). See installation instructions in the [operator README](cmd/operator/README.md#installation).
+
+Operator installs device plugins to the ```inteldeviceplugins-system``` namespace.
+
+## Install with HELM charts
+
+Device plugins can also be installed to a cluster using the device plugin [operator Helm chart](https://github.com/intel/helm-charts/tree/main/charts/device-plugin-operator) (depending on cert-manager and NFD). Individual plugin projects are under https://github.com/intel/helm-charts/tree/main/charts/.
+
+As with the method above, these steps will install device plugin operator and plugins under ```inteldeviceplugins-system``` namespace.
```
Comment (mythi) [inline INSTALL.md:19]: is it always the same namespace with helm or can you override with `helm install`?

Comment (tkatila) [reply INSTALL.md:19]: No, it's not hardcoded. If user chooses another namespace via **--namespace** then operator and plugins are installed under that namespace.

File: INSTALL.md (hunk 6 above)
Comment (tkatila) [reply INSTALL.md:13]: Sure. Though, I'd still add something like "Operator's default namespace is.." afterwards.

File: INSTALL.md (hunk 7 above)
Comment (tkatila) [reply INSTALL.md:52]: Added something, can you check if it's in the right direction?

Comment (mythi) [reply INSTALL.md:52]: looks OK, thanks!

**Review by mythi — DISMISSED** [review]
let's wait `e2e-gpu` to get back online

**Review by mythi — APPROVED** [review]

**Pull request merged**